lib/phase/functional.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+const Joi = require('joi');
+const Hoek = require('hoek');
+const clone = require('clone');
+
+// Actual environment size is limited by space, not quantity.
+// We do this to force sd.yaml to be simpler.
+const MAX_ENVIRONMENT_VARS = 25;
```

Comment:
Environment and Matrix are different.  Environment is static, Matrix is dynamic.  They both resolve to environment variables though.